azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/sync_clients.py

```diff
@@ -24,51 +29,17 @@ def __init__(self, transport):
         :param transport: The transport that the client will use.
         """
         self._transport = transport
-        self._transport.on_transport_connected = self._handle_transport_connected_state
-        self._transport.on_transport_disconnected = self._handle_transport_connected_state
+        self._transport.on_transport_connected = self._state_change
+        self._transport.on_transport_disconnected = self._state_change
 
         self.state = "initial"
+        self._message_queues = MessageQueueManager(
+            transport
```

Comment:
Have you considered using a simple dictionary (key/value pairs) with the queues? it seems like we're jumping through a lot of hoops to just enable/disable features when we could just be calling this on the queue objects directly.

Reply:
It was originally just a dictionary, yeah, but I expanded it to its own class since there was a lot of associated code cluttering up the client and I wanted to keep things clean and separated.

---
In reply to: [258281695](https://github.com/Azure/azure-iot-sdk-python-preview/pull/49#discussion_r258281695) [](ancestors = 258281695)